test/withRTLExtension_test.js

```diff
@@ -0,0 +1,151 @@
+      ));
+
+      expect(content).to.equal(expectedCSS);
+    });
+
+    it('handles match media queries', () => {
+      const styles = StyleSheet.create({
+        container: {
+          color: 'red',
+
+          '@media (min-width: 744px)': {
+            color: 'blue',
+          },
+
+          '@media (min-width: 1128px)': {
+            color: 'yellow',
+          },
+        },
+      });
+      const expectedCSS = '.container_ojhsnk{color:red !important;}@media (min-width: 744px){.container_ojhsnk{color:blue !important;}}@media (min-width: 1128px){.container_ojhsnk{color:yellow !important;}}';
+      const { css: { content } } = StyleSheetServer.renderStatic(() => (
+        ReactDOMServer.renderToString(<MyComponent styles={styles} />)
+      ));
+
+      expect(content).to.equal(expectedCSS);
+    });
+  });
+
+  describe('with _rtl style definition', () => {
+    it('handles regular styles', () => {
+      const styles = StyleSheet.create({
+        container: {
+          _rtl: {
+            color: 'red',
+          },
+        },
+      });
+      const expectedCSS = '[dir="rtl"] .container_1eedrr2{color:red !important;}';
+
+      const { css: { content } } = StyleSheetServer.renderStatic(() => (
+        ReactDOMServer.renderToString(<MyComponent styles={styles} />)
+      ));
+
+      expect(content).to.equal(expectedCSS);
+    });
+
+    it('handles pseudo selectors', () => {
+      const styles = StyleSheet.create({
+        container: {
+          _rtl: {
+            color: 'red',
+
+            ':before': {
+              color: 'blue',
+            },
+
+            ':after': {
+              color: 'yellow',
+            },
+          },
+        },
+      });
+      const expectedCSS = '[dir="rtl"] .container_1etudix{color:red !important;}[dir="rtl"] .container_1etudix:before{color:blue !important;}[dir="rtl"] .container_1etudix:after{color:yellow !important;}';
+      const { css: { content } } = StyleSheetServer.renderStatic(() => (
+        ReactDOMServer.renderToString(<MyComponent styles={styles} />)
+      ));
+
+      expect(content).to.equal(expectedCSS);
+    });
+
+    it('handles match media queries', () => {
+      const styles = StyleSheet.create({
+        container: {
+          _rtl: {
+            color: 'red',
+
+            '@media (min-width: 744px)': {
+              color: 'blue',
+            },
+
+            '@media (min-width: 1128px)': {
+              color: 'yellow',
+            },
+          },
+        },
+      });
+
+      const expectedCSS = '[dir="rtl"] .container_17kei9l{color:red !important;}@media (min-width: 744px){[dir="rtl"] .container_17kei9l{color:blue !important;}}@media (min-width: 1128px){[dir="rtl"] .container_17kei9l{color:yellow !important;}}';
+      const { css: { content } } = StyleSheetServer.renderStatic(() => (
+        ReactDOMServer.renderToString(<MyComponent styles={styles} />)
+      ));
+
+      expect(content).to.equal(expectedCSS);
+    });
+  });
+});
```

Comment:
I think Aphrodite also has support for some special things with keyframes and font faces. Can you also add some tests around these?

Reply:
hmm, i will dig into this. i'm not sure if things that are not custom selector handlers will need to be tested for.